### ISSUE TITLE
Format not supported?

### ISSUE THREAD

**Issue by vibes24** [octimot/StoryToolkitAI#98]
Windows 10 with 64gb of RAM, 3070 8gb.
Davinci Resolve Studio 18.1.4
StoryToolKitAI 17.16.2
Python 3.9.13

I tested the "MKV" video file with the standalone 17.16.2
The MKV file is 274MB, 1080p, H265, 22 minutes long.

First attempt without loading Resolve Studio.
The tool could not open/transcribe the video file.

**Second attempt:**
Load Resolve Studio - put in the video file on the timeline.
Click on transcribe timeline on StoryToolKit. It renders the video to "mov" format but then said format not supported and went to sleep.

**Third attempt:**
So in Resolve Studio, I exported the video file to WAVE sound format only. I checked the folder, it is exactly the same size (244MB as the MOV file earlier). 

I asked the standalone tool to transcribe the wave sound file. It fails.

Copy and paste the errors.

**Error 1 (second attempt, using resolve timeline)**
`INFO: Ignoring the new update (version 0.18.1) according to app settings.
INFO: Running StoryToolkitAI version 0.17.16.2 (standalone)
INFO: Using cuda for Torch / Whisper.
INFO: Resolve API module found and loaded
INFO: Resolve API module found and loaded
WARNING: Project settings directory C:\Users\vibes24\StoryToolkitAI\projects\Untitled Project 2023-04-22_125857 doesn't exist.
WARNING: Creating project settings directory.
INFO: Starting StoryToolkitAI GUI
INFO: Prompting user for render path.
INFO: Saving into C:/Users/vibes24/Documents/Transcripts from StoryToolKitAI and starting render.
INFO: Starting render
INFO: Starting to transcribe Timeline 1
INFO: Using cuda for Torch / Whisper.
INFO: Loading Whisper medium model.
INFO: Selected Whisper model is multilingual.
app.py:9843: UserWarning: PySoundFile failed. Trying audioread instead.
librosa\core\audio.py:184: FutureWarning: librosa.core.audio.__audioread_load
        Deprecated as of librosa version 0.10.0.
        It will be removed in librosa version 1.0.
  y, sr_native = __audioread_load(path, offset, duration, dtype)
ERROR: Traceback (most recent call last):
  File "librosa\core\audio.py", line 176, in load
    y, sr_native = __soundfile_load(path, offset, duration, dtype)
  File "librosa\core\audio.py", line 209, in __soundfile_load
    context = sf.SoundFile(path)
  File "soundfile.py", line 658, in __init__
  File "soundfile.py", line 1216, in _open
soundfile.LibsndfileError: Error opening 'C:\\Users\\vibes24\\Documents\\Transcripts from StoryToolKitAI/Timeline 1 1682182905.mov': Format not recognised.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "app.py", line 9337, in transcribe_from_queue
  File "app.py", line 9843, in whisper_transcribe
  File "librosa\core\audio.py", line 184, in load
    y, sr_native = __audioread_load(path, offset, duration, dtype)
  File "decorator.py", line 232, in fun
  File "librosa\util\decorators.py", line 60, in __wrapper
    return func(*args, **kwargs)
  File "librosa\core\audio.py", line 241, in __audioread_load
    reader = audioread.audio_open(path)
  File "audioread\__init__.py", line 132, in audio_open
    raise NoBackendError()
audioread.exceptions.NoBackendError

INFO: Transcription queue empty. Going to sleep.`

**Error 2: 3rd attempt using standalone tool to transcribe the wave file**

INFO: Ignoring the new update (version 0.18.1) according to app settings.
INFO: Running StoryToolkitAI version 0.17.16.2 (standalone)
INFO: Using cuda for Torch / Whisper.
INFO: Resolve API module found and loaded
INFO: Resolve API module found and loaded
WARNING: Resolve is not reachable. Retrying every few seconds.
INFO: Starting StoryToolkitAI GUI
INFO: Starting to transcribe test.wav
INFO: Using cuda for Torch / Whisper.
INFO: Loading Whisper medium model.
WARNING: Resolve is still not reachable. Now retrying every 15 seconds only a few more times.
INFO: Selected Whisper model is multilingual.
INFO: Detecting speech intervals in the audio.
INFO: Pre-detecting speech intervals in the audio.
Downloading: "https://github.com/snakers4/silero-vad/zipball/master" to C:\Users\vibes24/.cache\torch\hub\master.zip
2023-04-22 13:10:08.7941832 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '620'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.7981680 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '623'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8013993 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '625'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8046136 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '629'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8084611 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '628'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8117442 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '131'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8149895 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '134'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8181105 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '136'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8217527 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '140'. It is not used by any node and should be removed from the model.
2023-04-22 13:10:08.8273386 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '139'. It is not used by any node and should be removed from the model.
ERROR: Traceback (most recent call last):
  File "app.py", line 9337, in transcribe_from_queue
  File "app.py", line 9880, in whisper_transcribe
  File "app.py", line 9457, in get_speech_intervals
  File "app.py", line 9483, in combine_intervals
IndexError: list index out of range

INFO: Transcription queue empty. Going to sleep.
WARNING: Resolve not reachable after 15 tries. Disabling Resolve API polling until tool restart.

Thank you for your time.

(Edits to add information.)

**Comment by vibes24** [comment]
From the log file:

`Platform: Windows 10
 Platform version: 10.0.19045
 OS: 10 10.0.19045 SP0 Multiprocessor Free  ('', '', '')  
 running Python 3.9.13.final.0
-------------- (app.py:147)
2023-04-22 13:09:04,673 - StAI - DEBUG: All package requirements met. (app.py:167)
2023-04-22 13:09:04,674 - StAI - DEBUG: Loading config file C:\Users\ralph\StoryToolkitAI\config.json. (app.py:11553)
2023-04-22 13:09:04,881 - StAI - INFO: Ignoring the new update (version 0.18.1) according to app settings. (app.py:11797)
2023-04-22 13:09:04,882 - StAI - INFO: [1m[4mRunning StoryToolkitAI version 0.17.16.2 (standalone) (app.py:11370)
2023-04-22 13:09:04,910 - StAI - INFO: Using cuda for Torch / Whisper. (app.py:8737)
2023-04-22 13:09:04,910 - StAI - DEBUG: MotsResolve module initialized. (mots_resolve.py:42)
2023-04-22 13:09:04,922 - StAI - DEBUG: Trying to load DaVinciResolveScript module... (mots_resolve.py:159)
2023-04-22 13:09:04,923 - StAI - DEBUG: Found DaVinci Resolve at the default location: C:\Program Files\Blackmagic Design\DaVinci Resolve\ (mots_resolve.py:224)
2023-04-22 13:09:04,923 - StAI - DEBUG: Unable to find module DaVinciResolveScript from PYTHONPATH - trying default locations next (mots_resolve.py:236)
2023-04-22 13:09:04,943 - StAI - INFO: Resolve API module found and loaded (mots_resolve.py:258)
2023-04-22 13:09:06,992 - StAI - DEBUG: Trying to load DaVinciResolveScript module... (mots_resolve.py:159)
2023-04-22 13:09:07,093 - StAI - DEBUG: DaVinciResolveScript module loaded from PYTHONPATH (mots_resolve.py:173)
2023-04-22 13:09:07,094 - StAI - INFO: Resolve API module found and loaded (mots_resolve.py:258)
2023-04-22 13:09:09,125 - StAI - WARNING: Resolve is not reachable. Retrying every few seconds. (app.py:11125)
2023-04-22 13:09:09,153 - StAI - DEBUG: Running with TK 8.6.12 (app.py:3009)
2023-04-22 13:09:09,674 - StAI - INFO: Starting StoryToolkitAI GUI (app.py:3712)
2023-04-22 13:09:16,204 - StAI - DEBUG: Resolve is not started. (mots_resolve.py:297)
2023-04-22 13:09:16,204 - StAI - DEBUG: No Resolve project is loaded. (mots_resolve.py:313)
2023-04-22 13:09:16,205 - StAI - DEBUG: Resolve Media Pool not available. (mots_resolve.py:320)
2023-04-22 13:09:16,205 - StAI - DEBUG: Resolve Timeline not loaded or unavailable. (mots_resolve.py:324)
2023-04-22 13:09:16,205 - StAI - DEBUG: Resolve Bins not loaded or unavailable. (mots_resolve.py:328)
2023-04-22 13:09:28,186 - StAI - DEBUG: No transcription log window exists. (app.py:5919)
2023-04-22 13:09:50,207 - StAI - DEBUG: No transcription log window exists. (app.py:5919)
2023-04-22 13:09:50,212 - StAI - INFO: Starting to transcribe S24E08.wav (app.py:9325)
2023-04-22 13:09:50,212 - StAI - INFO: Using cuda for Torch / Whisper. (app.py:8737)
2023-04-22 13:09:50,239 - StAI - DEBUG: Closing window: ts-S24E08.wav-1682183368 (app.py:5234)
2023-04-22 13:09:50,251 - StAI - INFO: Loading Whisper medium model. (app.py:9816)
2023-04-22 13:10:00,566 - StAI - WARNING: Resolve is still not reachable. Now retrying every 15 seconds only a few more times. (app.py:11108)
2023-04-22 13:10:02,021 - StAI - INFO: Selected Whisper model is multilingual. (app.py:9833)
2023-04-22 13:10:07,419 - StAI - INFO: Detecting speech intervals in the audio. (app.py:9872)
2023-04-22 13:10:07,430 - StAI - INFO: Pre-detecting speech intervals in the audio. (app.py:9877)
2023-04-22 13:10:21,170 - StAI - ERROR: Traceback (most recent call last):
  File "app.py", line 9337, in transcribe_from_queue
  File "app.py", line 9880, in whisper_transcribe
  File "app.py", line 9457, in get_speech_intervals
  File "app.py", line 9483, in combine_intervals
IndexError: list index out of range
 (app.py:9342)
2023-04-22 13:10:21,184 - StAI - INFO: Transcription queue empty. Going to sleep. (app.py:9301)
2023-04-22 13:11:25,775 - StAI - WARNING: Resolve not reachable after 15 tries. Disabling Resolve API polling until tool restart. (app.py:11098)`

**Comment by vibes24** [comment]
I created a wav preset in resolve yet it continues to use MOV as the format for audio file.
![image](https://user-images.githubusercontent.com/60679996/233845949-5ef89ebe-33f7-4a48-a3c1-39f8e4a1cbbb.png)

INFO: Ignoring the new update (version 0.18.1) according to app settings.
INFO: Running StoryToolkitAI version 0.17.16.2 (standalone)
INFO: Using cuda for Torch / Whisper.
INFO: Resolve API module found and loaded
INFO: Resolve API module found and loaded
WARNING: Project settings directory C:\Users\vibes24\StoryToolkitAI\projects\Untitled Project 2023-04-23_102943 doesn't exist.
WARNING: Creating project settings directory.
INFO: Starting StoryToolkitAI GUI
INFO: Prompting user for render path.
INFO: Saving into C:/Users/ralph/Documents/Transcripts from StoryToolKitAI and starting render.
INFO: Starting render
INFO: Starting to transcribe Timeline 1
INFO: Using cuda for Torch / Whisper.
INFO: Loading Whisper medium model.
INFO: Selected Whisper model is multilingual.
app.py:9843: UserWarning: PySoundFile failed. Trying audioread instead.
librosa\core\audio.py:184: FutureWarning: librosa.core.audio.__audioread_load
        Deprecated as of librosa version 0.10.0.
        It will be removed in librosa version 1.0.
  y, sr_native = __audioread_load(path, offset, duration, dtype)
ERROR: Traceback (most recent call last):
  File "librosa\core\audio.py", line 176, in load
    y, sr_native = __soundfile_load(path, offset, duration, dtype)
  File "librosa\core\audio.py", line 209, in __soundfile_load
    context = sf.SoundFile(path)
  File "soundfile.py", line 658, in __init__
  File "soundfile.py", line 1216, in _open
soundfile.LibsndfileError: Error opening 'C:\\Users\\vibes24\\Documents\\Transcripts from StoryToolKitAI/Timeline 1 1682260241.mov': Format not recognised.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "app.py", line 9337, in transcribe_from_queue
  File "app.py", line 9843, in whisper_transcribe
  File "librosa\core\audio.py", line 184, in load
    y, sr_native = __audioread_load(path, offset, duration, dtype)
  File "decorator.py", line 232, in fun
  File "librosa\util\decorators.py", line 60, in __wrapper
    return func(*args, **kwargs)
  File "librosa\core\audio.py", line 241, in __audioread_load
    reader = audioread.audio_open(path)
  File "audioread\__init__.py", line 132, in audio_open
    raise NoBackendError()
audioread.exceptions.NoBackendError

INFO: Transcription queue empty. Going to sleep.

**Comment by octimot** [comment]
> I tested the "MKV" video file with the standalone 17.16.2
The MKV file is 274MB, 1080p, H265, 22 minutes long.
First attempt without loading Resolve Studio.
The tool could not open/transcribe the video file.

MKV containers are not supported unfortunately.

> Second attempt:
Load Resolve Studio - put in the video file on the timeline.
Click on transcribe timeline on StoryToolKit. It renders the video to "mov" format but then said format not supported and went to sleep.

What format and codec are you actually using for video and audio in this case?

> I created a wav preset in resolve yet it continues to use MOV as the format for audio file.

Have you un-checked "Export Video" in the "Video" tab?

**Comment by vibes24** [comment]
Hello. What about the "**Third Attempt**" on the first post where I actually extracted the audio to wave and tried to process it through the tool. It should be working since it is audio only in what I believe was supported format.

**Comment by octimot** [comment]
Sorry, I missed that!

I haven't seen this happening, except for the following cases:
- there's no actual audio or the audio is muted, 
- the audio file is not mono and the AI chooses to stick to one of the channels that don't have dialogue. 

Is this happening on other WAV files on your system? Could you try to run the tool on an audio file that is mono and you know for sure has audio (and dialogue, of course)?

You could also try to un-check the `Pre-Detect speech` option and run the transcription on a higher model (for eg. large-v2). The speech pre-detection AI might give false negatives if the audio is of lower quality, although I've never seen this happening...

**Comment by octimot** [comment]
Hey @vibes24 

Do you still need help on this issue? I'd close it if not, since we're unable to reproduce it here...

Cheers!

**Comment by vibes24** [comment]
Hey. Looks like it's user error. (My issues).
I was unsuccessful with Resolve multiple of times. When I tried MKV extract - I got AAC audio at 20mb. I tried Resolve again to convert to Wave but it was empty again. Using online converter from AAC to Wave made the StoryToolkitAI work for me. I've been busy but I will have to look into my audio exporting on Resolve Studio.
Thanks for your help and for this cool application.
